### PR TITLE
Use ICP-complaint AL2 repo URLs when building Docker images in China

### PR DESCRIPTION
**Pull request by tilne**
This commit modifies the process of building Docker images as part of
Batch cluster creation such that ICP-compliant URLs are used when using
the Amazon Linux 2 yum repos in China regions. This is accomplished by
setting yum variables used by the Amazon Linux 2 yum repo (but only when
building container images in China regions).

Signed-off-by: Tim Lane <tilne@amazon.com>

**Please See** [Git Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions)

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cli/src/pcluster/resources/batch/docker/scripts/modify_china_yum_vars.sh [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+main() {
+
+    if [ $# -ne 1 ]; then
+        error_exit_usage "Wrong number of arguments provided"
+    fi
+
+    # Only modify yum repo variables if cluster is running in a China region.
+    REGION="${1}"
+    if [[ "${REGION}" =~ ^cn-north-.* ]]; then
```
Comment (lukeseawalker) [inline cli/src/pcluster/resources/batch/docker/scripts/modify_china_yum_vars.sh:45]: what about matching also `cn-northwest-1`?

Comment (tilne) [reply cli/src/pcluster/resources/batch/docker/scripts/modify_china_yum_vars.sh:45]: ah yeah good catch and dumb mistake on my part. should be `^cn-.*`

Comment (tilne) [reply cli/src/pcluster/resources/batch/docker/scripts/modify_china_yum_vars.sh]: fixed

**Review by yuleiwan — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -21,14 +21,17 @@ CHANGELOG
...
 - Remove support for Python versions < 3.6.
 - Remove dependency on `future` package and `__future__` module.
 - Root volume size increased from 25GB to 35GB on all AMIs. Minimum root volume size is now 35GB.
-- Add sanity check to prevent cluster creation in non officially supported AWS regions 
+- Add sanity check to prevent cluster creation in non officially supported AWS regions
 - Change instance IAM policies to restrict run-instance API to only launching instances in the compute subnet.
 
+**BUG FIXES**
+
+- Use ICP-complaint AL2 repo URLs when building Docker images in China
```
Comment (mauri-melato) [inline CHANGELOG.md:34]: typo: compliant

Comment (tilne) [reply CHANGELOG.md:34]: Fixed in #2819

**Pull request merged**